### ISSUE TITLE
projects-test intermittently fails with "TimeoutError: Navigation Timeout Exceeded: 30000ms exceeded"

### ISSUE THREAD

**Issue by ecraig12345** [microsoft/fluentui#21191]
Northstar `projects-test` `rollup` test intermittently fails with this timeout error, on the last step where it's trying to load the built page. (I may have also seen other similar tests failing with the same error.) 

[example build with this failure](https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=231957&view=logs&j=258ec178-2d8b-5611-7b9b-60c5c95dae55&t=cc94c182-d6f4-5dae-734d-3676c12cee2a&l=11422)

```
verb @fluentui/projects-test test |  TimeoutError: Navigation Timeout Exceeded: 30000ms exceeded
verb @fluentui/projects-test test |      at /mnt/vss/_work/1/s/node_modules/puppeteer/lib/LifecycleWatcher.js:142:21
verb @fluentui/projects-test test |      at FrameManager.navigateFrame (/mnt/vss/_work/1/s/node_modules/puppeteer/lib/FrameManager.js:95:17)
verb @fluentui/projects-test test |      at Frame.goto (/mnt/vss/_work/1/s/node_modules/puppeteer/lib/FrameManager.js:407:12)
verb @fluentui/projects-test test |      at Page.goto (/mnt/vss/_work/1/s/node_modules/puppeteer/lib/Page.js:629:12)
verb @fluentui/projects-test test |      at performBrowserTest (/mnt/vss/_work/1/s/scripts/projects-test/performBrowserTest.ts:63:3)
verb @fluentui/projects-test test |      at rollup (/mnt/vss/_work/1/s/packages/fluentui/projects-test/src/rollup.ts:56:3)
verb @fluentui/projects-test test |      at performTest (/mnt/vss/_work/1/s/packages/fluentui/projects-test/src/index.ts:10:5)
```

For the rollup test it's this part of the test:
https://github.com/microsoft/fluentui/blob/18549da2abc46cee45243e6339969dac0d9e8cf7/packages/fluentui/projects-test/src/rollup.ts#L55-L56
Specifically this line:
https://github.com/microsoft/fluentui/blob/18549da2abc46cee45243e6339969dac0d9e8cf7/scripts/projects-test/performBrowserTest.ts#L63

It's not clear to me why this is happening or exactly how to debug. Ideas:
- different browser tests are running in parallel and one causes the other to fail? (but there's code earlier in performBrowserTest to ensure it's using a free port)
- output files aren't as expected / directory being served doesn't have expected contents? (could add logs to list/cat the output files)

**Comment by ecraig12345** [comment]
@TristanWatanabe If you're feeling better, could you help look into this? I think a good first step would be adding a log listing the files in the directory being served (and maybe their contents), and/or seeing if `express` has a verbose logging mode that can be enabled.

**Comment by ecraig12345** [comment]
Actually I think it's a race condition--search [this build log](https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=231957&view=logs&j=258ec178-2d8b-5611-7b9b-60c5c95dae55&t=cc94c182-d6f4-5dae-734d-3676c12cee2a&l=11422) for "Starting server on port" and you'll see that there are two different tests trying to run on port 8000. This is weird because I thought [portfinder](https://www.npmjs.com/package/portfinder) ([this line](https://github.com/microsoft/fluentui/blob/18549da2abc46cee45243e6339969dac0d9e8cf7/scripts/projects-test/performBrowserTest.ts#L26)) should have ensured the ports didn't collide? So we can either look into why `portfinder` isn't working, or try a different approach for ensuring each test uses a unique port.

**Comment by TristanWatanabe** [comment]
Reopening as this issue occurred once again after initial fix.

**Comment by Hotell** [comment]
run into another issue with `project-tests` project 

Error log:

```
================================================================================
ERR! test:projects:nextjs : ✔️ Temporary directories created under /tmp/project-nextjs--9272-6X4rujdRSD6G
ERR! ================================================================================
ERR! 
ERR! ================================================================================
ERR! test:projects:nextjs : STEP 1. Add dependencies to test project
ERR! ================================================================================
ERR! + cd /tmp/project-nextjs--9272-6X4rujdRSD6G/test-app &&
ERR!     yarn add next react@17 react-dom@17
ERR! info No lockfile found.
ERR! [1/4] Resolving packages...
ERR! [2/4] Fetching packages...
ERR! /mnt/vss/_work/1/s/.yarn/releases/cli.js:72706
ERR!               throw new Error(`Fetching/extracting of package ${params.url} seems to be hanging.`);
ERR!               ^
ERR! 
ERR! Error: Fetching/extracting of package https://registry.yarnpkg.com/@next/swc-linux-arm64-musl/-/swc-linux-arm64-musl-12.2.3.tgz seems to be hanging.
ERR!     at Timeout._onTimeout (/mnt/vss/_work/1/s/.yarn/releases/cli.js:72706:21)
ERR!     at listOnTimeout (internal/timers.js:557:17)
ERR!     at processTimers (internal/timers.js:500:7)
ERR! Error: child process exited with code 1
ERR!     at ChildProcess.<anonymous> (/mnt/vss/_work/1/s/scripts/gulp/sh.ts:29:14)
ERR!     at ChildProcess.emit (events.js:400:28)
ERR!     at maybeClose (internal/child_process.js:1058:16)
ERR!     at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
ERR! 
ERR! 
ERR! 
ERR! @fluentui/projects-test: The test suite failed, please check FAQ in "packages/fluentui/projects-test/README.md" for more details
ERR! 
ERR! 
ERR! 
ERR! ERR! code ELIFECYCLE
ERR! npm ERR! errno 1
ERR!  @fluentui/projects-test@0.63.1 test: `node -r @fluentui/scripts/babel/register src/index.ts`
ERR! npm ERR! Exit status 1

```
